### ISSUE TITLE
Unable to install socket.io on the intranet

### ISSUE THREAD

**Issue by msmolyak** [socketio/socket.io#2212]
When running

```
npm install socket.io --save
```

I am getting this error:

```
npm ERR fetch failed https://github.com/rase-/node-XMLHttpRequest/archive/a6b6f2.tar.gz
```

Apparently this archive is a dependency of the socket.io package. I am on the intranet with no connectivity to the Internet. How can I install socket.io? 

I can obtain the TAR file on the intranet, but where would I place it to satisfy the dependency? 

Also, is it possible to avoid building the module altogether by somehow obtaining a version pre-built for my platform (CentOS 6)?

**Comment by ichyr** [comment]
Windows 7, SP1, x64

Same issue. My intranet causes the following line after error in previous post:

`npm WARN retry will retry, erorr on last attempt: Error: certificate not trusted`

I've found a round way:
1. Navigate to where this dependency is located
2. Download this tarball
3. Install in manually.

Error will disappear. ***But this is not right way.***

**Comment by 1j01** [comment]
Are you [behind a proxy](http://jjasonclark.com/how-to-setup-node-behind-web-proxy/)?

**Comment by msmolyak** [comment]
I am on the intranet without access to the internet. I can import files from the internet, but the npm installation process does not have internet access.

**Comment by msmolyak** [comment]
ichyr, what do you mean to "install it manually". How do you install a TAR archive?

**Comment by ichyr** [comment]
Get tar archive to my remove machine and ( as here: https://docs.npmjs.com/cli/install )

```
npm install <tarball file>:

Install a package that is sitting on the filesystem. Note: if you just want to link a dev directory into your npm root, you can do this more easily by using npm link.

Example:

    npm install ./package.tgz
```

**Comment by msmolyak** [comment]
Hi ichyr,

I am able to install the XMLHttpRequest from the tarball, however this does not do anything to fix the original problem - npm install of socket.io still tries to download the a6.b6f2.tar.gz from GitHub. How did you stop npm form following the dependency to GitHub?

Thank you,

Michael

**Comment by ichyr** [comment]
I needed to install `karma` test runner. I've tried installing th elatest one, but resulted in this error we are talking about here. Thus asked guys from other project in my room.  I ended up installing version they use and it works: 0.10.10 ( or 0.10.9 )

You may find sources here to dig in the version socket.io references there.

https://github.com/karma-runner/karma/releases/tag/v0.10.9

**Comment by vladimir-ivanov** [comment]
Is this going to be fixed please? 

I am in the same situation, also webpack, karma-webpack and protractor fail for the same reason.
We are using corporate nexus instance to cache npm repos (common setup in multiple companies), but github.com dependencies can not be cached there.

Thanks

**Comment by vladimir-ivanov** [comment]
as a work around used shrinkwrap to fix the dependencies
https://gist.github.com/vladimir-ivanov/3c3ef070d73301ca3ccd

**Comment by yotamshapira** [comment]
AFAIK, this should be fixed now with the 1.4.0 release

**Comment by darrachequesne** [comment]
That issue was closed automatically. Please check if your issue is fixed with the latest release, and reopen if needed (with a [fiddle](https://github.com/darrachequesne/socket.io-fiddle) reproducing the issue if possible).
